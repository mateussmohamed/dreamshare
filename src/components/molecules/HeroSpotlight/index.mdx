---
name: HeroSpotlight
menu: Organisms
---

import { Playground, PropsTable } from 'docz'
import HeroSpotlight from './index'

# HeroSpotlight
<PropsTable of={HeroSpotlight} />

<Playground style={{ backgroundColor: 'grey', padding: 40 }}>
  <HeroSpotlight />
</Playground>
