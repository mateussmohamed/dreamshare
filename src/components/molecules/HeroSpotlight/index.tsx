import React from 'react'

import styled from 'styled-components'

import { Row, HeroTitle, HeroDescription, Button } from '../../atoms'

const HeroSpotlightContainer = styled(Row)`
  flex-direction: column;
  justify-content: center;
  align-items: center;
  max-width: 500px;
  text-align: center;

  @media only screen and (min-width: 320px) {
    margin: 20px auto;
    padding: 0 20px;

    button {
      width: 100%;
    }
  }

  @media only screen and (min-width: 768px) {
    margin: auto auto;
    padding: 0;
    button {
      width: auto;
    }
  }
`

export const HeroSpotlight = () => (
  <HeroSpotlightContainer>
    <HeroTitle>SHARE YOUR HOLIDAY DREAM</HeroTitle>
    <HeroDescription>
      And find the perfect partner to fullfill it
    </HeroDescription>
    <Button secondary>Find your holliday partner</Button>
  </HeroSpotlightContainer>
)
