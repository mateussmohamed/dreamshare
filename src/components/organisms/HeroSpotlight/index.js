import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import { Row, HeroTitle, HeroDescription } from 'atoms'
import { Button } from 'molecules'
// import { Row, HeroTitle, HeroDescription, Button } from 'components'

const HeroSpotlightStyled = styled(Row)`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  max-width: 500px;
  margin: auto auto;
  text-align: center;
`

const HeroSpotlight = props => (
  <HeroSpotlightStyled>
    <HeroTitle>{props.title}</HeroTitle>
    <HeroDescription>{props.description}</HeroDescription>
    <Button secondary>Find your holliday partner</Button>
  </HeroSpotlightStyled>
)

HeroSpotlight.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
}

export default HeroSpotlight
