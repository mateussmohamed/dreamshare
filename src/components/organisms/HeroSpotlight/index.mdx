---
name: HeroSpotlight
menu: Organisms
---

import { Playground, PropsTable } from 'docz'
import HeroSpotlight from './index'

# HeroSpotlight
<PropsTable of={HeroSpotlight} />

<Playground style={{ backgroundColor: 'grey', padding: 40 }}>
  <HeroSpotlight
    title="SHARE YOUR HOLIDAY DREAM"
    description="Find your holliday partner"
  />
</Playground>
