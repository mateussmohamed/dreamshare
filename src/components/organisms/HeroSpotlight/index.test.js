import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'

import HeroSpotlight from './index'

describe('[<HeroSpotlight />]', () => {
  const wrap = (props = {}) => shallow(<HeroSpotlight {...props} />).dive()

  test('matches to snapshot', () => {
    const wrapper = wrap({
      title: 'Hero Title',
      description: 'hero description',
    })

    expect(wrapper).toMatchSnapshot()
  })
})
