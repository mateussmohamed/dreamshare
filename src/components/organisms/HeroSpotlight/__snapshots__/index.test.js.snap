// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[<HeroSpotlight />] matches to snapshot 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "HeroSpotlight__HeroSpotlightStyled-j4g13-0",
        "isStatic": true,
        "rules": Array [
          "display:flex;flex-direction:column;justify-content:center;align-items:center;max-width:500px;margin:auto auto;text-align:center;",
        ],
      },
      "displayName": "HeroSpotlight__HeroSpotlightStyled",
      "render": [Function],
      "styledComponentId": "HeroSpotlight__HeroSpotlightStyled-j4g13-0",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
      Symbol(Symbol.hasInstance): [Function],
    }
  }
  forwardedRef={null}
>
  <HeroTitle>
    Hero Title
  </HeroTitle>
  <HeroDescription>
    hero description
  </HeroDescription>
  <Button
    secondary={true}
  >
    Find your holliday partner
  </Button>
</StyledComponent>
`;
